### PR TITLE
[RAPTOR-8243] stop using sagemaker sklearn extension package

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
`sagemaker_scikit_learn_extension` package is not up to date with `scikit-learn`.
Thus it has been removed from the scikit-learn public environment, and in the current PR it is removed from the examples and tests.

### CODE REVIEW

**Review by aslisabanci — APPROVED** [review]
LGTM as previously chatted with Yakov.

**Comment by yakov-g** [comment]
jarvis test cmrunner_integration

**Comment by yakov-g** [comment]
jarvis test cmrunner_integration

**Comment by yakov-g** [comment]
jarvis test cmrunner_integration

**Comment by yakov-g** [comment]
jenkins run cmrunner_integration

**Comment by yakov-g** [comment]
jarvis test cmrunner_integration

File: task_templates/3_pipelines/2_python3_sklearn_regression/create_pipeline.py [hunk 1]
```diff
@@ -92,10 +92,14 @@ def to_string(x):
 # For text, we:
 # 1. Impute missing values with the string "missing"
 # 2. Tfidf encode the text, using 1-grams and 2-grams.
+
+# Sklearn's TfidfVectorizer can only handle one column of text at a time,
+# and will fail on datastes with more than one text column.
```
Comment (akshoop) [inline task_templates/3_pipelines/2_python3_sklearn_regression/create_pipeline.py:97]: Typo:
```suggestion
# and will fail on datasets with more than one text column.
```

**Review by akshoop — APPROVED** [review]
LGTM from Core Modeling, CC @eric-s-s @danna-naser for awareness

File: task_templates/3_pipelines/6_python3_sklearn_multiclass/create_pipeline.py [hunk 2]
```diff
@@ -92,10 +92,14 @@ def to_string(x):
 # For text, we:
 # 1. Impute missing values with the string "missing"
 # 2. Tfidf encode the text, using 1-grams and 2-grams.
+
+# Sklearn's TfidfVectorizer can only handle one column of text at a time,
+# and will fail on datasets with more than one text column.
+# MultiColumnTfidfVectorizer from sagemaker-scikit-learn-extension may be useful to handle multiple text columns.
```
Comment (eric-s-s) [inline task_templates/3_pipelines/6_python3_sklearn_multiclass/create_pipeline.py:98]: add same note about impcompatibility: https://github.com/aws/sagemaker-scikit-learn-extension/issues/42

File: tests/fixtures/python3_sklearn_binary_hyperparameters/create_pipeline.py [hunk 3]
```diff
@@ -85,10 +85,14 @@ def text_selector(X):
 # For text, we:
 # 1. Impute missing values with the string "missing"
 # 2. Tfidf encode the text, using 1-grams and 2-grams.
+
+# Sklearn's TfidfVectorizer can only handle one column of text at a time,
+# and will fail on datasets with more than one text column.
+# MultiColumnTfidfVectorizer from sagemaker-scikit-learn-extension may be useful to handle multiple text columns.
```
Comment (eric-s-s) [inline tests/fixtures/python3_sklearn_binary_hyperparameters/create_pipeline.py:91]: and here

File: task_templates/3_pipelines/5_python3_sklearn_binary/create_pipeline.py [hunk 4]
```diff
@@ -93,10 +93,14 @@ def to_string(x):
...
 # 2. Tfidf encode the text, using 1-grams and 2-grams.
+
+# Sklearn's TfidfVectorizer can only handle one column of text at a time,
+# and will fail on datasets with more than one text column.
+# MultiColumnTfidfVectorizer from sagemaker-scikit-learn-extension may be useful to handle multiple text columns.
 text_pipeline = Pipeline(
     steps=[
         ("imputer", SimpleImputer(strategy="constant", fill_value="missing")),
-        ("tfidf", MultiColumnTfidfVectorizer(ngram_range=(1, 2))),
+        ("tfidf", TfidfVectorizer(ngram_range=(1, 2))),
```
Comment (eric-s-s) [inline task_templates/3_pipelines/5_python3_sklearn_binary/create_pipeline.py:103]: if we're not going to use it, please remove the text pipeline and have a comment that leads back to this PR

File: task_templates/3_pipelines/5_python3_sklearn_binary/create_pipeline.py [hunk 5]
```diff
@@ -105,7 +109,8 @@ def to_string(x):
     transformers=[
         ("num", numeric_pipeline, numeric_selector),
         ("cat", categorical_pipeline, categorical_selector),
-        ("txt", text_pipeline, text_selector),
+        # comment out because of TfidfVectorizer
```
Comment (eric-s-s) [inline task_templates/3_pipelines/5_python3_sklearn_binary/create_pipeline.py:112]: remove this and instead have a comment explaining that this was removed in this PR

**Review by eric-s-s — CHANGES_REQUESTED** [review]
please do not leave the old code in here.  instead have a comment explaining:
1. it was removed in this PR: link
2. why it was useful
3. why TFid can't replace it
4. why it was removed: issue link

File: task_templates/3_pipelines/2_python3_sklearn_regression/create_pipeline.py [hunk 6]
```diff
@@ -92,10 +92,14 @@ def to_string(x):
...
+# Sklearn's TfidfVectorizer can only handle one column of text at a time,
+# and will fail on datasets with more than one text column.
+# MultiColumnTfidfVectorizer from sagemaker-scikit-learn-extension may be useful to handle multiple text columns.
 text_pipeline = Pipeline(
     steps=[
         ("imputer", SimpleImputer(strategy="constant", fill_value="missing")),
-        ("tfidf", MultiColumnTfidfVectorizer(ngram_range=(1, 2))),
+        ("tfidf", TfidfVectorizer(ngram_range=(1, 2))),
     ]
 )
```
Comment (eric-s-s) [inline task_templates/3_pipelines/2_python3_sklearn_regression/create_pipeline.py:104]: please remove all this `text_pipeline` code.  make the comment above more clear with links (see request for changes)

**Comment by yakov-g** [comment]
> please do not leave the old text pipeline code in here. remove it completely. instead have a comment explaining:
> 
> 1. it was removed in this PR: link
> 2. why it was useful
> 3. why TFid can't replace it
> 4. why it was removed: issue link

Fixed

**Comment by yakov-g** [comment]
jenkins run pipelineController:all-expected-tasks

File: model_templates/python3_onnx_multiclass/preprocessing.py [hunk 7]
```diff
@@ -88,7 +94,7 @@ def text_selector(X):
 text_pipeline = Pipeline(
     steps=[
         ("imputer", SimpleImputer(strategy="constant", fill_value="missing")),
-        ("tfidf", MultiColumnTfidfVectorizer(ngram_range=(1, 2))),
+        ("tfidf", TfidfVectorizer(ngram_range=(1, 2))),
```
Comment (eric-s-s) [inline model_templates/python3_onnx_multiclass/preprocessing.py:97]: does this still work?

File: task_templates/3_pipelines/2_python3_sklearn_regression/create_pipeline.py [hunk 8]
```diff
@@ -89,22 +60,20 @@ def to_string(x):
...
         ("cat", categorical_pipeline, categorical_selector),
-        ("txt", text_pipeline, text_selector),
+        # Text preprocessing pipeline has been removed from here: https://github.com/datarobot/datarobot-user-models/pull/663
+        # Either TfidfVectorizer (scikit-learn) or MultiColumnTfidfVectorizer (sagemaker-scikit-learn-extension)
+        # can be used to process text.
+        # TfidfVectorizer can only handle one column of text at a time,
+        # and will fail on datasets with more than one text column.
+        # MultiColumnTfidfVectorizer may be useful to handle multiple text columns,
+        # but currently it is not compatible with scikit-learn:
+        # Checked with scikit-learn==0.24.2: ImportError: cannot import name 'VectorizerMixin' from 'sklearn.feature_extraction.text'
```
Comment (eric-s-s) [inline task_templates/3_pipelines/2_python3_sklearn_regression/create_pipeline.py:76]: link to the issue please

File: task_templates/3_pipelines/5_python3_sklearn_binary/create_pipeline.py [hunk 9]
```diff
@@ -90,22 +61,20 @@ def to_string(x):
...
         ("cat", categorical_pipeline, categorical_selector),
-        ("txt", text_pipeline, text_selector),
+        # Text preprocessing pipeline has been removed from here: https://github.com/datarobot/datarobot-user-models/pull/663
+        # Either TfidfVectorizer (scikit-learn) or MultiColumnTfidfVectorizer (sagemaker-scikit-learn-extension)
+        # can be used to process text.
+        # TfidfVectorizer can only handle one column of text at a time,
+        # and will fail on datasets with more than one text column.
+        # MultiColumnTfidfVectorizer may be useful to handle multiple text columns,
+        # but currently it is not compatible with scikit-learn:
+        # Checked with scikit-learn==0.24.2: ImportError: cannot import name 'VectorizerMixin' from 'sklearn.feature_extraction.text'
```
Comment (eric-s-s) [inline task_templates/3_pipelines/5_python3_sklearn_binary/create_pipeline.py:77]: link to the issue please

File: task_templates/3_pipelines/6_python3_sklearn_multiclass/create_pipeline.py [hunk 10]
```diff
@@ -92,10 +92,16 @@ def to_string(x):
...
+# MultiColumnTfidfVectorizer from sagemaker-scikit-learn-extension may be useful to handle multiple text columns,
+# but currently it is not compatible with scikit-learn:
+# Checked with scikit-learn==0.24.2: ImportError: cannot import name 'VectorizerMixin' from 'sklearn.feature_extraction.text'
 text_pipeline = Pipeline(
     steps=[
         ("imputer", SimpleImputer(strategy="constant", fill_value="missing")),
-        ("tfidf", MultiColumnTfidfVectorizer(ngram_range=(1, 2))),
+        ("tfidf", TfidfVectorizer(ngram_range=(1, 2))),
     ]
 )
```
[the remark this answers is not in the document]
Comment (eric-s-s) [reply task_templates/3_pipelines/6_python3_sklearn_multiclass/create_pipeline.py:106]: and rm https://github.com/datarobot/datarobot-user-models/pull/663/files#diff-ba6df9b01dc8fd73f0dae8915d7a17853150520ec23c515103b21e84ec7d8621R113

File: task_templates/3_pipelines/5_python3_sklearn_binary/create_pipeline.py [hunk 11]
```diff
@@ -90,22 +61,20 @@ def to_string(x):
...
-    ]
-)
 
 # Sparse preprocessing pipeline, for models such as Ridge that handle sparse input well
 sparse_preprocessing_pipeline = ColumnTransformer(
     transformers=[
         ("num", numeric_pipeline, numeric_selector),
         ("cat", categorical_pipeline, categorical_selector),
-        ("txt", text_pipeline, text_selector),
+        # Text preprocessing pipeline has been removed from here: https://github.com/datarobot/datarobot-user-models/pull/663
```
Comment (eric-s-s) [inline task_templates/3_pipelines/5_python3_sklearn_binary/create_pipeline.py:70]:
```suggestion
        # Text preprocessing pipeline has been removed: https://github.com/datarobot/datarobot-user-models/pull/663
```

File: task_templates/3_pipelines/2_python3_sklearn_regression/create_pipeline.py [hunk 12]
```diff
@@ -89,22 +60,20 @@ def to_string(x):
...
-    ]
-)
 
 # Sparse preprocessing pipeline, for models such as Ridge that handle sparse input well
 sparse_preprocessing_pipeline = ColumnTransformer(
     transformers=[
         ("num", numeric_pipeline, numeric_selector),
         ("cat", categorical_pipeline, categorical_selector),
-        ("txt", text_pipeline, text_selector),
+        # Text preprocessing pipeline has been removed from here: https://github.com/datarobot/datarobot-user-models/pull/663
```
Comment (eric-s-s) [inline task_templates/3_pipelines/2_python3_sklearn_regression/create_pipeline.py:69]:
```suggestion
        # Text preprocessing pipeline has been removed: https://github.com/datarobot/datarobot-user-models/pull/663
```

File: task_templates/3_pipelines/6_python3_sklearn_multiclass/create_pipeline.py [hunk 13]
```diff
@@ -92,10 +92,16 @@ def to_string(x):
 # For text, we:
 # 1. Impute missing values with the string "missing"
 # 2. Tfidf encode the text, using 1-grams and 2-grams.
+
+# Sklearn's TfidfVectorizer can only handle one column of text at a time,
```
Comment (eric-s-s) [inline task_templates/3_pipelines/6_python3_sklearn_multiclass/create_pipeline.py:96]: why does this still have a text_pipeline?  is there a reason that it's not the same as the other ones?

File: model_templates/python3_onnx_multiclass/preprocessing.py (hunk 7 above)
Comment (yakov-g) [reply model_templates/python3_onnx_multiclass/preprocessing.py:97]: Yes, this example uses skyserver dataset, which doesn't has only target field as text

File: task_templates/3_pipelines/6_python3_sklearn_multiclass/create_pipeline.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (yakov-g) [reply task_templates/3_pipelines/6_python3_sklearn_multiclass/create_pipeline.py:106]: this one fits on dataset with one text field, so it works

File: task_templates/3_pipelines/5_python3_sklearn_binary/create_pipeline.py (hunk 9 above)
Comment (yakov-g) [reply task_templates/3_pipelines/5_python3_sklearn_binary/create_pipeline.py:77]: why did i remove this? it should work

File: task_templates/3_pipelines/5_python3_sklearn_binary/create_pipeline.py (hunk 4 above)
Comment (yakov-g) [reply task_templates/3_pipelines/5_python3_sklearn_binary/create_pipeline.py:110]: This example also uses dataset with only one text field

**Comment by yakov-g** [comment]
jarvis run black_check

File: task_templates/3_pipelines/2_python3_sklearn_regression/create_pipeline.py [hunk 14]
```diff
@@ -92,10 +92,21 @@ def to_string(x):
 # For text, we:
 # 1. Impute missing values with the string "missing"
 # 2. Tfidf encode the text, using 1-grams and 2-grams.
+
+# Sklearn's TfidfVectorizer can only handle one column of text at a time,
```
Comment (eric-s-s) [inline task_templates/3_pipelines/2_python3_sklearn_regression/create_pipeline.py:96]: this comment is exactly the same as below.  please remove it.  I'm assuming that this works with the current dataset?

File: task_templates/3_pipelines/5_python3_sklearn_binary/create_pipeline.py [hunk 15]
```diff
@@ -93,10 +93,21 @@ def to_string(x):
 # For text, we:
 # 1. Impute missing values with the string "missing"
 # 2. Tfidf encode the text, using 1-grams and 2-grams.
+
+# Sklearn's TfidfVectorizer can only handle one column of text at a time,
```
Comment (eric-s-s) [inline task_templates/3_pipelines/5_python3_sklearn_binary/create_pipeline.py:97]: rm duplicate comment

File: task_templates/3_pipelines/6_python3_sklearn_multiclass/create_pipeline.py (hunk 13 above)
Comment (eric-s-s) [reply task_templates/3_pipelines/6_python3_sklearn_multiclass/create_pipeline.py:96]: assuming it works, rm duplicate comment

File: task_templates/3_pipelines/5_python3_sklearn_binary/create_pipeline.py (hunk 15 above)
Comment (yakov-g) [reply task_templates/3_pipelines/5_python3_sklearn_binary/create_pipeline.py:97]: done

File: task_templates/3_pipelines/2_python3_sklearn_regression/create_pipeline.py (hunk 14 above)
Comment (yakov-g) [reply task_templates/3_pipelines/2_python3_sklearn_regression/create_pipeline.py:96]: done

File: tests/fixtures/python3_sklearn_binary_hyperparameters/create_pipeline.py [hunk 16]
```diff
@@ -82,22 +53,19 @@ def text_selector(X):
...
         ("cat", categorical_pipeline, categorical_selector),
-        ("txt", text_pipeline, text_selector),
+        # Text preprocessing pipeline has been removed from here: https://github.com/datarobot/datarobot-user-models/pull/663
+        # Either TfidfVectorizer (scikit-learn) or MultiColumnTfidfVectorizer (sagemaker-scikit-learn-extension)
+        # can be used to process text.
+        # TfidfVectorizer can only handle one column of text at a time,
+        # and will fail on datasets with more than one text column.
+        # MultiColumnTfidfVectorizer may be useful to handle multiple text columns,
+        # but currently it is not compatible with scikit-learn:
+        # Checked with scikit-learn==0.24.2: ImportError: cannot import name 'VectorizerMixin' from 'sklearn.feature_extraction.text'
```
Comment (eric-s-s) [inline tests/fixtures/python3_sklearn_binary_hyperparameters/create_pipeline.py:68]: :sob: issue!

Comment (eric-s-s) [reply tests/fixtures/python3_sklearn_binary_hyperparameters/create_pipeline.py:68]: once the issue link is added, this is good to go

File: tests/fixtures/python3_sklearn_binary_hyperparameters/create_pipeline.py [hunk 17]
```diff
@@ -82,22 +53,19 @@ def text_selector(X):
...
-    ]
-)
-
 # Sparse preprocessing pipeline, for models such as Ridge that handle sparse input well
 sparse_preprocessing_pipeline = ColumnTransformer(
     transformers=[
         ("num", numeric_pipeline, numeric_selector),
         ("cat", categorical_pipeline, categorical_selector),
-        ("txt", text_pipeline, text_selector),
+        # Text preprocessing pipeline has been removed from here: https://github.com/datarobot/datarobot-user-models/pull/663
```
Comment (eric-s-s) [inline tests/fixtures/python3_sklearn_binary_hyperparameters/create_pipeline.py:61]:
```suggestion
        # Text preprocessing pipeline has been removed: https://github.com/datarobot/datarobot-user-models/pull/663
```

File: tests/fixtures/python3_sklearn_binary_hyperparameters/create_pipeline.py (hunk 16 above)
Comment (yakov-g) [reply tests/fixtures/python3_sklearn_binary_hyperparameters/create_pipeline.py]: added

**Review by eric-s-s — APPROVED** [review]

**Comment by yakov-g** [comment]
jenkins run  black_check

**Comment by yakov-g** [comment]
jenkins run pipelineController:all-expected-tasks

**Comment by yakov-g** [comment]
jenkins run black_check

**Comment by yakov-g** [comment]
jenkins run black_check

**Comment by yakov-g** [comment]
jenkins run pipelineController:all-expected-tasks

**Comment by yakov-g** [comment]
jenkins run continuous-integration/jenkins/pr-merge

**Comment by yakov-g** [comment]
jenkins run  pipelineController:all-expected-tasks

**Pull request merged**